packages/@aws-cdk/aws-codepipeline-actions/test/manual-approval.test.ts

```diff
@@ -25,6 +26,123 @@ nodeunitShim({
       test.done();
     },
 
+    'allows granting manual approval permissions to role'(test: Test) {
+      const stack = new Stack();
+      const role = new iam.Role(stack, 'Human', { assumedBy: new iam.AnyPrincipal() });
+      const manualApprovalAction = new cpactions.ManualApprovalAction({
+        actionName: 'Approve',
+      });
+      const pipeline = new codepipeline.Pipeline(stack, 'pipeline');
+      pipeline.addStage({
+        stageName: 'Source',
+        actions: [new cpactions.GitHubSourceAction({
+          actionName: 'Source',
+          output: new codepipeline.Artifact(),
+          oauthToken: SecretValue.plainText('secret'),
+          owner: 'aws',
+          repo: 'aws-cdk',
+        })],
+      });
+      const stage = pipeline.addStage({ stageName: 'stage' });
+      stage.addAction(manualApprovalAction);
+
+      manualApprovalAction.grantManualApproval(role);
+
+      expect(stack).to(haveResourceLike('AWS::IAM::Policy', {
+        'PolicyDocument': {
+          'Statement': [
+            {
+              'Action': 'codepipeline:ListPipelines',
+              'Effect': 'Allow',
+              'Resource': '*',
+            },
+            {
+              'Action': [
+                'codepipeline:GetPipeline',
+                'codepipeline:GetPipelineState',
+                'codepipeline:GetPipelineExecution',
+              ],
+              'Effect': 'Allow',
+              'Resource': {
+                'Fn::Join': [
+                  '',
+                  [
+                    'arn:',
+                    {
+                      'Ref': 'AWS::Partition',
+                    },
+                    ':codepipeline:',
+                    {
+                      'Ref': 'AWS::Region',
+                    },
+                    ':',
+                    {
+                      'Ref': 'AWS::AccountId',
+                    },
+                    ':',
+                    {
+                      'Ref': 'pipelineDBECAE49',
+                    },
+                  ],
+                ],
```

Comment:
Let's conserve the lines here a little bit more:

```suggestion
                'Fn::Join': ['', [
                  'arn:',
                  {'Ref': 'AWS::Partition' },
                  ':codepipeline:',
                  {'Ref': 'AWS::Region' },
                  ':',
                  {'Ref': 'AWS::AccountId' },
                  ':',
                  {'Ref': 'pipelineDBECAE49'},
                ]],
```

Same comment below.